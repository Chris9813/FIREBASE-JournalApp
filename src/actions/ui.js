import { types } from "../types/types";

export const setError = (err) => ({
    type: types.uiSetError,
    payload: err,
})

export const remError = (err) => ({
    type: types.uiRemoveError,
})

