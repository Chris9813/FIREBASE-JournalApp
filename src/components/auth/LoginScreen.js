import React from 'react'


import { useDispatch, useSelector } from 'react-redux'
import { Link } from 'react-router-dom'
import { startGoogleLogin, startLoginEmailPaswword } from '../../actions/auth'
import { useForm } from '../hooks/useForm'


export const LoginScreen = () => {

    const {loading} = useSelector(state => state.ui)
    
    const dispatch = useDispatch();
    
    
    const [formValues, handleInputChange ] = useForm({
        email: "chris9813@gmail.com",
        password: "123456"
    });

    const {email, password} = formValues

    const handleLogin = (e) => {
        e.preventDefault()
        dispatch( startLoginEmailPaswword(email, password ));
    }

    const handleGoogleLogin = () => {
        dispatch(startGoogleLogin());
    }


    return (
        <div className="auth_main">
            <div className="auth_box-container">
            <h3 className ="auth_title">Login</h3>
            <form onSubmit={handleLogin}
            className= "animate__animated animate__fadeIn animate__faster"
            >
                <input
                className="aut_input"
                type="text"
                placeholder="email"
                name="email"
                autoComplete="off"
                value = {email}
                onChange={handleInputChange}
                />

                <input
                className="aut_input"
                type="password"
                placeholder="password"
                name="password"
                value={password}
                onChange={handleInputChange}
                />

                <button
                type="submit"
                className = "btn btn-primary btn-block"
                disabled={loading}
                >
                    Login
                </button>


        <div className="auth_social-networks">
            <p>Login with social networks</p>
            
            
            <div 
                className="google-btn"
                onClick= {handleGoogleLogin}
            >
                <div className="google-icon-wrapper">
                    <img className="google-icon" src="https://upload.wikimedia.org/wikipedia/commons/5/53/Google_%22G%22_Logo.svg" alt="google button" />
                </div>
                <p className="btn-text">
                    <b>Sign in with google</b>
                </p>
            </div>


        </div>
        <Link 
        className="link"
        to = "/auth/register">
            create new account
        </Link>
        </form>
            </div>
            
    </div>
    )
}
