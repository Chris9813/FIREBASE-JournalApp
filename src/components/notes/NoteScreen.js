import React, { useEffect, useRef } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { activeNote, startDeleting } from '../../actions/notes'
import { useForm } from '../hooks/useForm'
import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {
    const dispatch = useDispatch();

    const {active:note} = useSelector(state => state.notes)
    const [values, handleInputChange, reset] = useForm(note)
    
    const {body, title, id} = values

    const activeId = useRef( note.id )

    useEffect(() => {
        if(note.id !== activeId.current){
            reset(note)
            activeId.current = note.id
        }
    }, [note, reset])

    useEffect(() => {
        dispatch(activeNote(values.id, {...values}))
        
    }, [values, dispatch])

    const handleDelete = () => {
        dispatch(startDeleting(id));
    }

    return (
        <div className="notes_main-content">
            <NotesAppBar />

            <div className="notes_content">
                <input 
                type="text"
                placeholder="Another awsome title"
                className="notes_title-input"
                autoComplete="off"
                value = {title}
                onChange = {handleInputChange}
                name="title"
                />

                <textarea
                placeholder="What happened today"
                className="notes_textarea"
                value={body}
                onChange={handleInputChange}
                name="body"
                >
                </textarea>

                {
                note.url &&
                <div className="notes_image">
                    <img 
                    src={note.url}
                    alt="Imagen"
                    height= "150px"
                    />
                </div>
                }
            </div>
            <button className="btn btn-danger"
            onClick={handleDelete}
            >
                    Delete
            </button>
        </div>
    )
}
