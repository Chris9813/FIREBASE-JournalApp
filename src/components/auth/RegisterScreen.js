import React from 'react'
import { Link } from 'react-router-dom'
import { useForm } from '../hooks/useForm'
import validator from "validator"
import { useDispatch, useSelector } from 'react-redux'
import { remError, setError } from '../../actions/ui'
import { startRegisterWithEmailPasswordName } from '../../actions/auth'

export const RegisterScreen = () => {

    const dispatch = useDispatch();

    const {msgError} = useSelector(state => state.ui)


    const[values, handleInputChange] = useForm({
        name: "Chris",
        email: "chris9813@gmail.com",
        password: "123456",
        password2: "123456",
    })
    
    const {name, email, password, password2} = values

    const handleRegister = (e) => {
        e.preventDefault();
        if(isFormValid()) {
            dispatch(startRegisterWithEmailPasswordName(email, password, name))
        }
    }

    const isFormValid = () => {
            if(name.trim().length === 0){
                dispatch(setError("Name is Required"))
                return false
            } else if (!validator.isEmail(email)) {
                dispatch(setError("Email Incorrect"))
                return false
            } else if (password !== password2 || password.length < 5) {
                dispatch(setError("password Incorrect"))
                return false
            }
        dispatch(remError(""))
        return true
    }


    return (
        <div className="auth_main">
            <div className="auth_box-container">
            <div className= "animate__animated animate__fadeIn animate__faster">
            <h3 className ="auth_title">Register</h3>
            <form onSubmit = {handleRegister}>

            {
            msgError && 
            (<div className="auth_alert-error">
            {msgError}
            </div>)
            }

            <input
                className="aut_input"
                type="text"
                placeholder="Email"
                name="email"
                autoComplete="off"
                onChange={handleInputChange}
                value={email}
                />

                <input
                className="aut_input"
                type="text"
                placeholder="Name"
                name="name"
                autoComplete="off"
                onChange={handleInputChange}
                value={name}
                />

                <input
                className="aut_input"
                type="password"
                placeholder="password"
                name="password"
                onChange={handleInputChange}
                value={password}
                />

                <input
                className="aut_input"
                type="password"
                placeholder="password2"
                name="password2"
                onChange={handleInputChange}
                value={password2}
                />

                <button
                type="submit"
                className = "btn btn-primary btn-block"
                onClick={handleRegister}
                >
                    Register
                </button>
        <Link 
        className="link"
        to = "/auth/login">
            Already register
        </Link>
        </form>
    </div>
            </div>
            
        </div>
    )
}
