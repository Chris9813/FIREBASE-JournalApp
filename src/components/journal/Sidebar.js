import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { startLogout } from '../../actions/auth';
import { noteLogout, startNewNote } from '../../actions/notes';
import { JournalEntries } from './JournalEntries';


export const Sidebar = () => {

    const dispatch = useDispatch();

    const handleLogout = () => {
        dispatch(startLogout());
        dispatch(noteLogout());
    }
    
    const {name} = useSelector(state => state.auth)
    const nameSort = name.split(" ")[0]

    const handleAddNew = () => {
        dispatch(startNewNote());
    }

    return (
        <div className="journal_sidebar">
            <div className = "journal_sidebar-navbar">
                <h3 className="mt-5">
                    <i className ="far fa-moon"></i>
                    <span>{nameSort}</span>
                </h3>
                <button
                className="btn"
                onClick={handleLogout}
                >Logout</button>
                
            </div>
            <div className="journal_new-entry"
                onClick={handleAddNew}
            >
                <i className="far fa-calendar-plus fa-5x"></i>
            <p
            className="mt-5"
            >New entry</p>
            </div>
            <JournalEntries />
        </div>
    )
}
