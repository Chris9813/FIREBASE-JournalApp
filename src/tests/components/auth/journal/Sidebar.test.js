import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import { configure, mount } from "enzyme";
import React from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import { startLogout } from "../../../../actions/auth";
import { noteLogout, startNewNote } from "../../../../actions/notes";
import { Sidebar } from "../../../../components/journal/Sidebar";



jest.mock("../../../../actions/auth", () => ({
    startLogout: jest.fn(),
}));

jest.mock("../../../../actions/notes", () => ({
    noteLogout: jest.fn(),
    startNewNote: jest.fn(),
}));


const middlewares = [thunk];
const mockStore = configureStore(middlewares);

configure({ adapter: new Adapter() });

const initState = {
    auth: {name: 'chris'},
    ui: {
        loading: false,
        msgError: null,
    },
    notes: {
        active:{
            id: "hola",
        },
        notes: [],
    }
};

let store = mockStore(initState)
store.dispatch = jest.fn();

describe('Pruebas en <SideBar />', () => {
    
    beforeEach(() => {
        store = mockStore(initState);
        jest.clearAllMocks();
    });

    const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <Sidebar />
        </MemoryRouter>
    </Provider>
    );

    test('debe de mostrarse correctaente ', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('debe llamar la accion starLogout ', () => {
        wrapper.find('.btn').prop('onClick')();
        expect(startLogout).toHaveBeenCalled();
        expect(noteLogout).toHaveBeenCalled();
    });

    test('debe de llamar el startNewNote ', () => {
        wrapper.find('.journal_new-entry').prop('onClick')();
        expect(startNewNote).toHaveBeenCalled();
        
    });

});
