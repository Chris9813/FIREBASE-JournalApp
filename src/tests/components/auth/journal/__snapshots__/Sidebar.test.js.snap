// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <SideBar /> debe de mostrarse correctaente  1`] = `
<div
  className="journal_sidebar"
>
  <div
    className="journal_sidebar-navbar"
  >
    <h3
      className="mt-5"
    >
      <i
        className="far fa-moon"
      />
      <span>
        chris
      </span>
    </h3>
    <button
      className="btn"
      onClick={[Function]}
    >
      Logout
    </button>
  </div>
  <div
    className="journal_new-entry"
    onClick={[Function]}
  >
    <i
      className="far fa-calendar-plus fa-5x"
    />
    <p
      className="mt-5"
    >
      New entry
    </p>
  </div>
  <div
    className="journal_entries "
  />
</div>
`;
